dialdbg/src/main.rs

```diff
@@ -0,0 +1,239 @@
+mod parse;
+mod strings;
+
+use anyhow::{anyhow, Result};
+use clap::Parser;
+use log4rs::append::file::FileAppender;
+use log4rs::config::{Appender, Config, Root};
+use serde_json::{from_str, Value};
+use std::{fs, io, path::PathBuf};
+use tokio::time::Instant;
+use viam::rpc::dial::{self, ViamChannel};
+use webrtc::stats;
+
+/// dialdbg gives information on how to connect to a Viam robot
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+struct Args {
+    /// Filepath for remote configuration JSON containing URI and secret
+    #[arg(short, long)]
+    config: Option<PathBuf>,
+
+    /// Whether direct gRPC connection should not be examined
+    #[arg(long, action, conflicts_with("nowebrtc"))]
+    nogrpc: bool,
+
+    /// Whether WebRTC connection should not be examined
+    #[arg(long, action, conflicts_with("nogrpc"))]
+    nowebrtc: bool,
+
+    /// Filepath for output of dialdbg (file will be overwritten)
+    #[arg(short, long)]
+    output: Option<PathBuf>,
+
+    /// Secret with which to connect to the robot
+    #[arg(short, long, required_unless_present("config"))]
```

Comment:
[minor] we should allow this to work with no secret

Reply:
Done.